src/main.ts

```diff
@@ -251,8 +269,33 @@ export default class NewZettel extends Plugin {
     } else {
       titleContent = "";
     }
-    const fullContent = titleContent + content;
-    const file = await this.app.vault.create(path, fullContent);
+
+    let file = null;
+    const backlinkRegex = /{{link}}/g;
+    const titleRegex = /{{title}}/g;
+    if (this.settings.templateFile && this.settings.templateFile != "") {
+      let template_content = "";
+      try {
+        template_content = await this.app.vault.adapter.read(this.settings.templateFile);
+      } catch (err) {
+        new Notice(`Couldn't read template file ${this.settings.templateFile} make sure that it is a valid file...`);
+        return;
+      }
+
+      if (!titleRegex.test(template_content) || !backlinkRegex.test(template_content)) {
+        new Notice("Title {{title}} or Backlink {{link}} placeholder in template missing. Please add and try again...");
```

Comment:
Instead of just throwing a notice let's add a setting of if we want to throw an error or just silently continue with the default. Also I think the default should be to silently continue. Assume not everyone will have a template.

Reply:
I check beforehand ([This Line | Will Also Comment Separately](https://github.com/Dyldog/luhman-obsidian-plugin/pull/43/commits/2e1cf2b9a4aef2f137558a1f4247292f838226f5#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR276)), if the template setting is empty, so basically saying you don't have a template selected or chosen to not use one, then it will go use the previous version of note creation ([Here | Will Comment As Well](https://github.com/Dyldog/luhman-obsidian-plugin/pull/43/commits/2e1cf2b9a4aef2f137558a1f4247292f838226f5#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR293)), so notice will be triggered there ever.

Now If a template is set, non-empty setting, it will check first if the file is readable ([Here | Will Comment As Well](https://github.com/Dyldog/luhman-obsidian-plugin/pull/43/commits/2e1cf2b9a4aef2f137558a1f4247292f838226f5#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR278)) and if not throw an error notice that is informing the user they might have a non-existent path to a template set or other file reading problems that might occur. After that, it will check, if the two placeholders that exist are present and if not, telling the user the template is malformed (The Code Marked for Change). Since without both of them or at least without the title, the file/note will be problematic afterwards for the search function since it won't guarantee the title is an H1 which is from my understanding used for the search/open command of the plugin.

So I would say this should stay because it will inform the user there is something wrong with the template they are using, if they are using a template at all. Though, I would say I should add a more detailed description to the setting informing the user that this is the case, and they are required to use both of the placeholders.

Reply:
@Eforen